torch/csrc/jit/codegen/cuda/kernel_ir.cpp

```diff
@@ -114,6 +114,57 @@ UpdateMagicZero::UpdateMagicZero(IrBuilderPasskey passkey)
       "IR type only valid for Kernel container.");
 }
 
+namespace {
+
+bool isIntegralScalar(const Val* val) {
+  return val->isScalar() && val->getDataType().has_value() &&
+      isIntegralType(val->getDataType().value());
+}
+
+} // namespace
+
+IntPair::IntPair(IrBuilderPasskey passkey)
+    : Val(passkey, ValType::IntPair, DataType::Int) {}
```

Comment:
Should `DataType` be `Index` instead of `Int`?

Reply:
Yes makes sense. Changed to `Index`. Thanks.